packages/smooth_app/lib/pages/product/common/product_list_page.dart

```diff
@@ -331,6 +334,16 @@ class _ProductListPageState extends State<ProductListPage>
                     : appLocalizations.product_could_not_remove,
               ),
               duration: SnackBarDuration.medium,
+              action: SnackBarAction(
+                  textColor: PRIMARY_BLUE_COLOR,
+                  label: appLocalizations.undo,
+                  onPressed: () async {
+                    barcodes.insert(index, removedItem);
+                    _selectedBarcodes.add(barcode);
+                    productList.set(barcodes);
+                    await DaoProductList(localDatabase).put(productList);
+                    setState(() {});
+                  }),
```

Comment:
Could be problematic as higher in the code you consider that the `remove` may not work.
Be careful, especial because here the list are supposed to contain unique barcodes.

Reply:
I am not adding any duplicate barcodes, whatever has been removed from the list will be added back to it on pressed !!
I did not quite understand what you are conveying

Reply:
```dart
final bool removed = productList.remove(barcode);
if (removed) {
```
means that we test if the barcode was actually removed from the productList, obviously.
But in the rest of the code you just supposed that the product was indeed removed.
That's why I suggest that if `removed` is false, you just return.
Then, on your snack bar you can put the barcode back, you're safe as you've checked it's been removed before.

Reply:
Oh gotcha ! Thanks for the explanation 😄! Will make the changes as suggested!